src/plugin/systems/worlds.rs

```diff
@@ -0,0 +1,111 @@
+//! Systems responsible for interfacing our Bevy components with the Rapier physics engine.
+
+use crate::dynamics::{
+    RapierImpulseJointHandle, RapierMultibodyJointHandle, RapierRigidBodyHandle,
+};
+use crate::geometry::RapierColliderHandle;
+use crate::plugin::RapierContext;
+use crate::prelude::PhysicsWorld;
+use bevy::prelude::*;
+
+/// If an entity is turned into the child of something with a physics world, the child should become a part of that physics world
+///
+/// If this fails to happen, weirdness will ensue.
+pub fn on_add_entity_with_parent(
+    q_add_entity_without_parent: Query<(Entity, &Parent), Changed<Parent>>,
```

Comment:
I imagine we should filter for any handles or existing PhisicsWorld, this query risks iterating through a lot of entities (probably especially UI)

Reply:
I absolutely agree, I believe when I made that I was just worried about missing any child hierarchy changes. IE: If this entity's children were in the physics simulation, I would want to change its children to reflect the new world they should be a part of. I don't think this is an especially common usecase, I just didn't want any unexpected functionality to be present.